consensus/istanbul/backend/backend.go

```diff
@@ -172,15 +186,32 @@ func (sb *backend) Commit(proposal istanbul.Proposal, seals [][]byte) error {
 	}
 
 	h := block.Header()
-	// Append seals into extra-data
-	err := writeCommittedSeals(h, seals)
-	if err != nil {
-		return err
+
+	// Append Round Number and seals into extra-data of qbft consensus
+	if sb.IsQIBFTConsensus() {
+		err := writeQBFTCommittedSeals(h, seals)
+		if err != nil {
+			return err
+		}
+		err = writeRoundNumber(h, round)
+		if err != nil {
+			return err
+		}
+	} else {
+		// Append seals into extra-data
+		err := writeCommittedSeals(h, seals)
+		if err != nil {
+			return err
+		}
 	}
+
+	// Remove ValidatorSet added to ProposerPolicy registry, if not done, the registry keeps increasing size with each block height
+	sb.config.ProposerPolicy.ClearRegistry()
+
 	// update block's header
 	block = block.WithSeal(h)
 
-	sb.logger.Info("Committed", "address", sb.Address(), "hash", proposal.Hash(), "number", proposal.Number().Uint64())
+	sb.logger.Info("QBFT: Committed", "address", sb.Address(), "hash", proposal.Hash(), "number", proposal.Number().Uint64())
```

Comment:
might not be QBFT, right? depends on `sb.IsQIBFTConsensus()`

Reply:
Right. I pushed a commit to fix the log